src/main/java/nl/windesheim/codeparser/analyzers/util/FilePartResolver.java

```diff
@@ -0,0 +1,84 @@
+package nl.windesheim.codeparser.analyzers.util;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Node;
+import nl.windesheim.codeparser.FilePart;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Resolves FileParts.
+ */
+public final class FilePartResolver {
+
+    /**
+     *
+     */
+    private FilePartResolver() {
+
+    }
+
+    /**
+     * @param node The node you want the FilePart of
+     * @return the FilePart of the node
+     */
+    public static FilePart getFilePartOfNode(final Node node) {
+
+        Node currentNode = node;
+        while (!(currentNode instanceof CompilationUnit)) {
+            if (!currentNode.getParentNode().isPresent()) {
+                break;
+            }
+            currentNode = currentNode.getParentNode().get();
+        }
+
+        //Set the ClassOrInterface part for the strategy context
+        if (currentNode instanceof CompilationUnit) {
+            if (((CompilationUnit) currentNode).getStorage().isPresent()) {
+                String filename = ((CompilationUnit) currentNode).getStorage().get().getFileName();
+                File file = new File(filename);
+
+                if (node.getRange().isPresent()) {
```

Comment:
3 geneste ifjes, graag voorkomen

Zie https://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting